rascal2/dialogs/project_dialog.py

```diff
@@ -0,0 +1,209 @@
+        self.project_name_label = QtWidgets.QLabel("Project Name:", self)
+        self.project_name_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignLeft)
+        self.project_name_label.setStyleSheet(self._label_style)
+
+        self.project_name = QtWidgets.QLineEdit(self)
+        self.project_name.setPlaceholderText("Enter project name")
+
+        self.project_name_error = QtWidgets.QLabel("Project name needs to be specified.", self)
+        self.project_name_error.setStyleSheet(self._error_style)
+        self.project_name_error.hide()
+
+        # Project folder labels
+        self.project_folder_label = QtWidgets.QLabel("Project Folder:", self)
+        self.project_folder_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignLeft)
+        self.project_folder_label.setStyleSheet(self._label_style)
+
+        self.project_folder = QtWidgets.QLineEdit(self)
+        self.project_folder.setReadOnly(True)
+        self.project_folder.setPlaceholderText("Select project folder")
+
+        self.project_folder_error = QtWidgets.QLabel("An empty project folder needs to be selected.", self)
+        self.project_folder_error.setStyleSheet(self._error_style)
+        self.project_folder_error.hide()
+
+    def create_buttons(self) -> None:
+        """
+        Create buttons.
+        """
+        self.browse_button = QtWidgets.QPushButton(" Browse", self)
+        self.browse_button.setIcon(QtGui.QIcon(path_for("browse-light.png")))
+        self.browse_button.clicked.connect(self.open_folder_selector)
+        self.browse_button.setStyleSheet(self._button_style.format("#403F3F"))
+
+        self.create_button = QtWidgets.QPushButton(" Create", self)
+        self.create_button.setIcon(QtGui.QIcon(path_for("create.png")))
+        self.create_button.clicked.connect(self.create_project)
+        self.create_button.setStyleSheet(self._button_style.format("#0D69BB"))
+
+        self.cancel_button = QtWidgets.QPushButton(" Cancel", self)
+        self.cancel_button.setIcon(QtGui.QIcon(path_for("cancel.png")))
+        self.cancel_button.clicked.connect(self.cancel_project_creation)
+        self.cancel_button.setStyleSheet(self._button_style.format("#E34234"))
+
+    def cancel_project_creation(self) -> None:
+        """
+        Cancel project creation.
+        """
+        self.close() if self.parent().new_project_action_called else self.parent().toggleView()
+        self.reset_variables()
+
+    def reset_variables(self) -> None:
+        """
+        Reset variables.
+        """
+        self.project_folder.setText("")
+        self.project_name.setText("")
+        self.name_error = False
+        self.folder_error = False
+
+    def open_folder_selector(self) -> None:
+        """
+        Open folder selector.
+        """
+        folder_path = QtWidgets.QFileDialog.getExistingDirectory(self, "Select Folder")
+        if folder_path:
+            files_in_folder = list(filter(lambda x: x[0] != ".", os.listdir(folder_path)))
```

Comment:
i think this'd be clearer if it used a list comprehension instead:
`files_in_folder = [file for files in os.listdir(folder_path) if not file.startswith(".")]`
but if you want to keep it this way then it should still use `not x.startswith(".")` rather than `x[0] != "."` for clarity